src/SourceBuild/content/test/Microsoft.DotNet.Tests/PackageSourceMappingsTests.cs

```diff
@@ -0,0 +1,242 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Xml.Linq;
+using Microsoft.DotNet.UnifiedBuild.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.Tests
+{
+    [Trait("Category", "PackageSourceMappings")]
+    public class PackageSourceMappingsTests
+    {
+        private static readonly PackageSourceMappingsSetup TestSetup = PackageSourceMappingsSetup.Instance;
+        private ITestOutputHelper OutputHelper { get; }
+
+        public PackageSourceMappingsTests(ITestOutputHelper outputHelper)
+        {
+            OutputHelper = outputHelper;
+        }
+
+        // Unified build - with existing mappings - online
+        [Fact]
+        public void UnifiedBuildWithMappings()
+        {
+            string[] sources = ["source-built-arcade", "source-built-runtime", "net-sdk-supporting-feed"];
+            RunTest("ub-mappings.config", true, sources, customSources: ["net-sdk-supporting-feed"], sourceBuild: false);
+        }
+
+        // Unified build - with mappings - online - no local sources
+        [Fact]
+        public void UnifiedBuildWithMappingsNoLocalSources()
+        {
+            string[] sources = ["net-sdk-supporting-feed"];
+            RunTest("ub-mappings-nolocal.config", true, sources, customSources: ["net-sdk-supporting-feed"], sourceBuild: false);
+        }
+
+        // Unified build - no mappings - online
+        [Fact]
+        public void UnifiedBuildNoMappings()
+        {
+            string[] sources = ["source-built-arcade", "source-built-runtime", "net-sdk-supporting-feed"];
```

Comment:
This test's implementation appears identical to UnifiedBuildWithMappings except for the config file.  For the SourceBuildTests you utilized a theory.  Why not the same here?   Overall, I wonder if a single Theory wouldn't make the tests simpler.

Reply:
Will fix - thanks! I missed collapsing these two tests into a common Theory.

Reply:
Fixed with https://github.com/dotnet/sdk/pull/44649/commits/f13f4f1cfb5a7adec023571eaf49ddcc8a051648

I have not tried to merge all tests into a single Theory. While possible, I feel that it would not bring much value, and make inline data harder to manage.